### PR TITLE
Add register type to Modbus climate docs

### PR DESCRIPTION
**Pull request by vzahradnik**
## Proposed change

This change adds a new optional parameter `current_temp_register_type` into `Modbus climate` to allow reading values from both input and holding registers.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: home-assistant/home-assistant#31944
- This PR fixes or closes issue:

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_integrations/climate.modbus.markdown [hunk 1]
```diff
@@ -32,6 +32,7 @@ climate:
     temp_step: 1
     target_temp_register: 2782
     current_temp_register: 27586
+    current_temp_register_type: holding
```
Comment (frenck) [inline source/_integrations/climate.modbus.markdown]: We try to keep all configuration samples minimal. Thus, no optional requirement in the default sample. This helps a user to get started quickly by copy-&-paste the sample without worrying about optional parameters which they most likely not need. If required, insert a full configuration sample later that covers special setups or alike.

File: source/_integrations/climate.modbus.markdown [hunk 2]
```diff
@@ -56,6 +57,10 @@ current_temp_register:
   description: Register number for current temperature (Process value).
   required: true
   type: integer
+current_temp_register_type:
```
Comment (frenck) [inline source/_integrations/climate.modbus.markdown:59]: the `default: ` is missing, which it has, according to the PR on the codebase.

**Review by frenck — CHANGES_REQUESTED** [review]

File: source/_integrations/climate.modbus.markdown (hunk 1 above)
Comment (vzahradnik) [reply source/_integrations/climate.modbus.markdown]: OK, I will update that. Thanks!

**Review by frenck — APPROVED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

**Pull request merged**